lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -617,10 +617,15 @@ else if(!request.keyName.equals(keyName))
 
 		/* timestamp */
 		if(request.timestamp == 0) {
-			if(options.queryTime)
-				request.timestamp = ably.time();
-			else
+			if (options.queryTime) {
+				/* Spec: RSA10k */
+				if (timeOffset == null) {
+					timeOffset = ably.time() - timestamp();
+				}
+				request.timestamp = timeOffset + timestamp();
+			} else {
 				request.timestamp = timestamp();
```

Comment:
> 3) if new request with queryTime = 1 initiated we need to get synced with server time once again?

Yes, although to be clear that's not obvious in the spec whether it should query again or not if queryTime is explicitly set.  However, the point is that if queryTime is set true, then it gets the server time and offset, and then uses that offset for all time calculations moving forwards as in most conditions that will suffice.